pallets/aliases/src/lib.rs

```diff
@@ -0,0 +1,410 @@
+pub use pallet::*;
+use pallet_clusters::Clusters;
+use pallet_fragments::GetInstanceOwnerParams;
+use pallet_protos::{Proto, ProtoOwner, Protos};
+use sp_clamor::{Hash128, Hash256, InstanceUnit};
+use sp_std::vec::Vec;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod dummy_data;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+/// Enum that indicates the types of target assets linked to an alias
+#[derive(Encode, Decode, Clone, scale_info::TypeInfo, Debug, PartialEq)]
+pub enum LinkTarget<TAccountId> {
+	/// A Proto defined by its hash
+	Proto(Hash256),
+	/// A Fragment instance defined by
+	Fragment {
+		/// definition_hash of the Fragment target
+		definition_hash: Vec<u8>,
+		/// edition_id of the Fragment target
+		edition_id: InstanceUnit,
+		/// copy_id of the Fragment target
+		copy_id: InstanceUnit }, // struct variant allows a nicer UX on polkadotJS
+	/// An AccountId
+	Account(TAccountId),
+	/// A Cluster defined by its hash
+	Cluster(Hash128),
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::{
+		pallet_prelude::*,
+		sp_runtime::{traits::Zero, SaturatedConversion},
+	};
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config
+		+ pallet_balances::Config
+		+ pallet_protos::Config
+		+ pallet_fragments::Config
+		+ pallet_clusters::Config
+	{
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The price (in native NOVA) for a Namespace
+		#[pallet::constant]
+		type NamespacePrice: Get<u128>;
+
+		/// The max size of namespace and alias strings
+		#[pallet::constant]
+		type NameLimit: Get<u32>;
+	}
+
+	/// **StorageMap** that maps a **Namespace** to its owner.
+	#[pallet::storage]
+	pub type Namespaces<T: Config> = StorageMap<_, Twox64Concat, Vec<u8>, T::AccountId>;
+
+	/// **StorageMap** that maps a **Name (of type `Vec<u8>`)** to an **index**.
+	/// This ensures no duplicated aliases are used.
+	#[pallet::storage]
+	pub type Names<T: Config> = StorageMap<_, Twox64Concat, Vec<u8>, Compact<u64>>;
+
+	/// **StorageValue** that **equals** the **total number of unique names** used for aliases in the chain.
+	#[pallet::storage]
+	pub type NamesIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	/// **StorageDoubleMap** that maps a (**Namespace**, **Alias** name index) with a **LinkTarget**.
+	#[pallet::storage]
+	pub type Aliases<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		Vec<u8>, // namespace
+		Twox64Concat,
+		Compact<u64>,             // alias name index
+		LinkTarget<T::AccountId>, // target asset
```

Comment:
@alebaffa you are using the enum with the wrong `Vec<u8>` here.... how can you say its not related to storage? I'm confused.